### PR TITLE
refactor(lacchain.system): add missing ricardian contracts

### PR DESCRIPTION
**Pull request by leisterfrancisco**
Update Ricardian Contract Content

## Change Description
Add new Ricardian content to:
- Warning, action `onerror` does not have a ricardian contract
- Warning, action `addentity` does not have a ricardian contract
- Warning, action `rmentity` does not have a ricardian contract
- Warning, action `addvalidator` does not have a ricardian contract
- Warning, action `addwriter` does not have a ricardian contract
- Warning, action `addboot` does not have a ricardian contract
- Warning, action `addobserver` does not have a ricardian contract
- Warning, action `netaddgroup` does not have a ricardian contract
- Warning, action `rmnode` does not have a ricardian contract
- Warning, action `netrmgroup` does not have a ricardian contract
- Warning, action `netsetgroup` does not have a ricardian contract
- Warning, action `setnodeinfo` does not have a ricardian contract
- Warning, action `setnodexinfo` does not have a ricardian contract
- Warning, action `setentinfo` does not have a ricardian contract
- Warning, action `setentxinfo` does not have a ricardian contract
- Warning, action `setschedule` does not have a ricardian contract
- Warning, action `setram` does not have a ricardian contract
- Warning, action `onblock` does not have a ricardian contract
- Warning, action `reindex` does not have a ricardian contract

### CODE REVIEW

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+
+On error action, notification of this action is delivered to the sender of a deferred transaction
+when an objective error occurs while executing the deferred transaction.
+
+<h1 class="contract">addentity</h1>
+
+---
+spec_version: "0.2.0"
+title: Add Entity
+summary: 'Add to the network a Partner and Non Partner entity'
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Register a partner or non-partner entity on the network

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to Add to the network a Partner and Non Partner entity.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows the user to register a partner or non-partner entity on the network

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+
+<h1 class="contract">rmentity</h1>
+
+---
+spec_version: "0.2.0"
+title: Remove Entity
+summary: 'Remove from the network a Partner and Non Partner entity'
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Remove a partner or non-partner entity from the network

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to remove from the network a Partner and Non Partner entity.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows the user to remove a partner or non-partner entity from the network.

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+---
+
+It is executed with each block and passes some data that can be used for network monitoring and as an effect trigger.
+
+<h1 class="contract">reindex</h1>
+
+---
+spec_version: "0.2.0"
+title: Dummy Action
+summary: 'I'm a dummy action!'
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md:395]: dummy action

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+I'm a dummy action!.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Dummy action for testing and development puroposes.

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+<h1 class="contract">setram</h1>
+
+---
+spec_version: "0.2.0"
+title: Set Ram Resource
+summary: 'Set ram resource to an account'
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to set ram resource to an account.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: RAM

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 8]
```diff
@@ -0,0 +1,399 @@
...
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Set schedule action, sets a new list of active validators, by proposing a schedule change, once the block that contains the proposal becomes irreversible, the schedule is promoted to "pending" automatically. Once the block that promotes the schedule is irreversible, the schedule will become "active".
+
+<h1 class="contract">setram</h1>
+
+---
+spec_version: "0.2.0"
+title: Set Ram Resource
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: RAM (caps)

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 9]
```diff
@@ -0,0 +1,399 @@
...
+<h1 class="contract">setentxinfo</h1>
+
+---
+spec_version: "0.2.0"
+title: Set Extended Entity Information
+summary: 'Set extended information to an Entity'
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to set extendend information to an entity.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: please spell check:  extendend

I believe this action can only be performed by the permisioning committee
 please check required auth for this action

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 10]
```diff
@@ -0,0 +1,399 @@
...
+<h1 class="contract">setentinfo</h1>
+
+---
+spec_version: "0.2.0"
+title: Set Entity Information
+summary: 'Set information to an Entity'
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to set information to an entity.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows an entity to set information such as entity name, location, and contact information.

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 11]
```diff
@@ -0,0 +1,399 @@
...
+---
+
+Allows the user to set extendend information to a specific node.
+
+<h1 class="contract">setentinfo</h1>
+
+---
+spec_version: "0.2.0"
+title: Set Entity Information
+summary: 'Set information to an Entity'
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Set information about an Entity

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 12]
```diff
@@ -0,0 +1,399 @@
...
+---
+
+Allows the user to remove from the network a Partner and Non Partner entity.
+
+<h1 class="contract">addvalidator</h1>
+
+---
+spec_version: "0.2.0"
+title: Add Validator
+summary: 'Add to the network a Validator for the entity'
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Register a validator node on the network

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 13]
```diff
@@ -0,0 +1,399 @@
...
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to Add to the network a Validator for the entity.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows a partner entity to register a new validator node on the network.

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 14]
```diff
@@ -0,0 +1,399 @@
...
+
+<h1 class="contract">addwriter</h1>
+
+---
+spec_version: "0.2.0"
+title: Add Writer
+summary: 'Add to the network a Writer node for a Partner entity'
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Register a writer node on the network

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 15]
```diff
@@ -0,0 +1,399 @@
...
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to Add to the network a Writer node for a Partner entity.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows a partner or non-partner entity to register a new writer node on the network.

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 16]
```diff
@@ -0,0 +1,399 @@
...
+
+<h1 class="contract">addboot</h1>
+
+---
+spec_version: "0.2.0"
+title: Add Boot Node
+summary: 'Add a Boot Node to the network'
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Register a boot node on the network

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 17]
```diff
@@ -0,0 +1,399 @@
...
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to add a Boot Node to the network.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows a partner entity to register a new boot node on the network.

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 18]
```diff
@@ -0,0 +1,399 @@
...
+
+<h1 class="contract">addobserver</h1>
+
+---
+spec_version: "0.2.0"
+title: Add Observer Node
+summary: 'Add a Observer Node to the network'
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Register an observer node on the network

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 19]
```diff
@@ -0,0 +1,399 @@
...
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to add a Observer Node to the network.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows a partner or non-partner entity to register a new observer node on the network.

**Review by xavier506 — CHANGES_REQUESTED** [review]
@leisterfrancisco  please see my comments and make sure you apply a spell check.

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 20]
```diff
@@ -0,0 +1,399 @@
...
+
+<h1 class="contract">netaddgroup</h1>
+
+---
+spec_version: "0.2.0"
+title: Create a Group of Nodes
+summary: 'Create a Group of Nodes to the network'
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Create a group to organize nodes on the network

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 21]
```diff
@@ -0,0 +1,399 @@
...
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to add a Group of Nodes to the network.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows the permissioning comittee to create a group for nodes on the network.

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 22]
```diff
@@ -0,0 +1,399 @@
...
+<h1 class="contract">rmnode</h1>
+
+---
+spec_version: "0.2.0"
+title: Remove a Node
+summary: 'Remove a Node from the network'
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to remove a Node from the network.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows an entity to remove a Node from the network.

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 23]
```diff
@@ -0,0 +1,399 @@
...
+<h1 class="contract">netrmgroup</h1>
+
+---
+spec_version: "0.2.0"
+title: Remove a Group of Nodes
+summary: 'Remove a Group of Nodes from the network'
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to remove a Group of Nodes from the network.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows the permisioning committee

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 24]
```diff
@@ -0,0 +1,399 @@
...
+<h1 class="contract">netsetgroup</h1>
+
+---
+spec_version: "0.2.0"
+title: Link a Node
+summary: 'Link a Node with a list of Groups to the network'
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to link a Node with a list of Groups to the network.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: I believe it is the permisioning committee that can call net set  group

Comment (xavier506) [reply contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: it is also linking groups with groups , not nodes with groups (please check)

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 25]
```diff
@@ -0,0 +1,399 @@
...
+<h1 class="contract">setnodeinfo</h1>
+
+---
+spec_version: "0.2.0"
+title: Set Node Information
+summary: 'Set information to a Node'
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to set information to a specific node.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows an entity to set information about a specific node on the network.

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 26]
```diff
@@ -0,0 +1,399 @@
...
+<h1 class="contract">setnodexinfo</h1>
+
+---
+spec_version: "0.2.0"
+title: Set Node Extended Information
+summary: 'Set extendend information to a Node'
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the user to set extendend information to a specific node.
```
Comment (xavier506) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Allows the permissioning committee to set extended information about a specific node  

please check require auth in contract action to verify this

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md (hunk 9 above)
Comment (leisterfrancisco) [reply contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: No, It doesn't but `setnodeinfo` does

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md (hunk 26 above)
Comment (leisterfrancisco) [reply contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: According to Smart Contract Action, It allows both, Partner and Non Partner Entities

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md (hunk 24 above)
Comment (leisterfrancisco) [reply contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: The action header
image

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md [hunk 27]
```diff
@@ -0,0 +1,399 @@
...
+<h1 class="contract">setnodexinfo</h1>
+
+---
+spec_version: "0.2.0"
+title: Set Node Extended Information
+summary: 'Set extended information to a {{nowrap node}} Node'
+icon: @ICON_BASE_URL@/@ADMIN_ICON_URI@
+---
+
+Allows the permissioning committee to set {{ext_info}} extended information to a specific {{node}} node.
```
Comment (leisterfrancisco) [inline contracts/lacchain.system/ricardian/lacchain.system.contracts.md:332]:
```
void lacchain::setnodexinfo(const name& node, const std::string& ext_info) {

   node_table nod es(get_self(), get_self().value);
   auto itr_node = nodes.find( node.value );
   eosio::check(itr_node != nodes.end(), "node does not exists");

   require_auth( get_self() );

   nodes.modify( itr_node, eosio::same_payer, [&]( auto& row ) {
      row.ext_info = ext_info;
   });
}
```

@xavier506  I don't think so it is necessary the permissioning committee

File: contracts/lacchain.system/ricardian/lacchain.system.contracts.md (hunk 9 above)
Comment (leisterfrancisco) [reply contracts/lacchain.system/ricardian/lacchain.system.contracts.md]: Correction, functions makes a `get_self()` which before a I didn't know that eosio is the permissioning committee so then

**Review by xavier506 — APPROVED** [review]
LGTM

**Pull request merged**